vhostuser/vhostuser.go

```diff
@@ -199,16 +200,21 @@ func cmdAdd(args *skel.CmdArgs) error {
 
 	createVhostPort(n, args.ContainerID)
 
-	// run the IPAM plugin and get back the config to apply
-	result, err = ipam.ExecAdd(n.IPAM.Type, args.StdinData)
-	if err != nil {
-		return fmt.Errorf("failed to set up IPAM: %v", err)
-	}
-	if result.IP4 == nil {
-		return errors.New("IPAM plugin returned missing IPv4 config")
+	if n.IPAM.Type != "" {
+		// run the IPAM plugin and get back the config to apply
+		result, err = ipam.ExecAdd(n.IPAM.Type, args.StdinData)
+		if err != nil {
+			return fmt.Errorf("failed to set up IPAM: %v", err)
+		}
+		if result.IP4 == nil {
+			return errors.New("IPAM plugin returned missing IPv4 config")
+		}
+
+		containerIP = result.IP4.IP.IP.String()
+	} else {
+		containerIP = "0.0.0.0"
```

Comment:
@kangasta The IP address 0.0.0.0 has a different meaning. FYI https://www.lifewire.com/four-zero-ip-address-818384. Is there a specific reason to but container IP 0.0.0.0? Can you please little bit explain on the use case of using this address in your implementation ?

Reply:
@kangasta Shouldn't be we just send back the empty result.Print() as in SRIOV CNI - 	https://github.com/Intel-Corp/sriov-cni/blob/master/sriov/sriov.go#L608

Reply:
Yes, good point. 

The 0.0.0.0 was only used to avoid `"ipAddr": "/32"` line in IP4 conf file, but it would probably be better to skip the whole `SetupContainerNetwork(...)` call to avoid writing the IP4 config with incomplete IP data and to skip calling the vhost tool config with unspecified IP.

Reply:
+1. IP should be empty when IPAM config is empty.